### PR TITLE
BAEL-4842 - Use React and Spring Boot to Build a Simple CRUD App

### PR DESCRIPTION
**Pull request by saleco**
Initial source module for the article

### CODE REVIEW

File: spring-boot-modules/spring-boot-react/src/main/java/com/baeldung/springbootreact/controller/ClientsController.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+public class ClientsController {
+
+    private final ClientRepository clientRepository;
+
+    public ClientsController(ClientRepository clientRepository) {
+        this.clientRepository = clientRepository;
+    }
+
+    @GetMapping
+    public List getClients() {
```
Comment (KevinGilmore) [inline spring-boot-modules/spring-boot-react/src/main/java/com/baeldung/springbootreact/controller/ClientsController.java:23]:
```suggestion
    public List<Client> getClients() {
```

File: spring-boot-modules/spring-boot-react/src/main/java/com/baeldung/springbootreact/controller/ClientsController.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        return clientRepository.findById(id).orElseThrow(RuntimeException::new);
+    }
+
+    @PostMapping
+    public ResponseEntity createClient(@RequestBody Client client) throws URISyntaxException {
+        Client savedClient = clientRepository.save(client);
+        return ResponseEntity.created(new URI("/clients/" + savedClient.getId())).body(savedClient);
+    }
+
+    @PutMapping("{id}")
```
Comment (KevinGilmore) [inline spring-boot-modules/spring-boot-react/src/main/java/com/baeldung/springbootreact/controller/ClientsController.java:38]:
```suggestion
    @PutMapping("/{id}")
```

**Review by KevinGilmore — COMMENTED** [review]
Please see recommended changes

File: spring-boot-modules/spring-boot-react/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+### Relevant Articles:
+
+- [Use React and Spring Boot to Build a Simple CRUD App](https://www.baeldung.com/use-react-and-spring-boot-to-build-a-simple-crud-app/)
```
[the remark this answers is not in the document]
Comment (saleco) [reply spring-boot-modules/spring-boot-react/README.md]: Done

**Pull request merged**